src/Microbenchmarks.jl

```diff
@@ -0,0 +1,75 @@
+module Microbenchmarks
+
+using DataFrames, CSV, StatsBase
+
+function read_bench(benchfile::String)
+    # Load benchmark data from file
+    benchmarks = CSV.read(benchfile, DataFrame; header=["language", "benchmark", "time"])
+
+    # Capitalize and decorate language names from datafile
+    dict = Dict("c"=>"C", "julia"=>"Julia", "lua"=>"LuaJIT", "fortran"=>"Fortran", "java"=>"Java",
+                "javascript"=>"JavaScript", "matlab"=>"Matlab", "mathematica"=>"Mathematica",
+                "python"=>"Python", "octave"=>"Octave", "r"=>"R", "rust"=>"Rust", "go"=>"Go");
+    benchmarks[!,:language] = [dict[lang] for lang in benchmarks[!,:language]]
+
+    # Normalize benchmark times by C times
+    ctime = benchmarks[benchmarks[!,:language].== "C", :]
+    benchmarks = innerjoin(benchmarks, ctime, on=:benchmark, makeunique=true)
+    select!(benchmarks, Not([:language_1]))
+    rename!(benchmarks, :time_1 =>:ctime)
+    benchmarks[!,:normtime] = benchmarks[!,:time] ./ benchmarks[!,:ctime];
+
+    # Compute the geometric mean for each language
+    langs = [];
+    means = [];
+    priorities = [];
+    for lang in unique(benchmarks[!,:language])
+        data = benchmarks[benchmarks[!,:language].== lang, :]
+        gmean = geomean(data[!,:normtime])
+        push!(langs, lang)
+        push!(means, gmean)
+        if (lang == "C")
+            push!(priorities, 1)
+        elseif (lang == "Julia")
+            push!(priorities, 2)
+        else
+            push!(priorities, 3)
+        end
+    end
+
+    # Add the geometric means back into the benchmarks dataframe
+    langmean = DataFrame(language=langs, geomean = means, priority = priorities)
+    benchmarks = innerjoin(benchmarks, langmean, on=:language)
+
+    # Put C first, Julia second, and sort the rest by geometric mean
+    sort!(benchmarks, [:priority, :geomean]);
+    sort!(langmean,   [:priority, :geomean]);
+
+    return benchmarks
+end
+
+benchmarks = read_bench("benchmarks.csv")
+
+#=
+p = plot(benchmarks,
+         x = :language,
```

Comment:
A lot of the circles hide each other. I wonder whether we can make better use of the space on the x-axis. Ideally, if the circles collide, we'd move one of them slightly to the left or to the right to avoid the collision.

As a simplification, maybe we can systematically or randomly put them at slightly different positions? If we'd do it systematically, we could also make it easier to distinguish the colors. Would something like
`x = :language .+ range(-0.3, 0.3, length(benchmarks))`
look better or worse? The 0.3 might need some adjustment.

Reply:
I prefer to currently leave it is as is for this PR. @asinghvi17 has a better plot that we'll pull in soon.